src/middleware/context-storage/index.ts

```diff
@@ -0,0 +1,55 @@
+/**
+ * @module
+ * Context Storage Middleware for Hono.
+ */
+
+import type { Context } from '../../context'
+import type { Env, MiddlewareHandler } from '../../types'
+import { AsyncLocalStorage } from 'node:async_hooks'
+
+const asyncLocalStorage = new AsyncLocalStorage<Context>()
+
+/**
+ * Context Storage Middleware for Hono.
+ *
+ * @see {@link https://hono.dev/docs/middleware/builtin/context-storage}
+ *
+ * @returns {MiddlewareHandler} The middleware handler function.
+ *
+ * @example
+ * ```ts
+ * type Env = {
+ *   Variables: {
+ *     message: string
+ *   }
+ * }
+ *
+ * const app = new Hono<Env>()
+ *
+ * app.use(contextStorage())
+ *
+ * app.use(async (c, next) => {
+ *   c.set('message', 'Hono is cool!!)
+ *   await next()
+ * })
+ *
+ * app.get('/', async (c) => { c.text(getMessage()) })
+ *
+ * const getMessage = () => {
+ *   return getContext<Env>().var.message
+ * }
+ * ```
+ */
+export const contextStorage = (): MiddlewareHandler => {
+  return async (c, next) => {
```

Comment:
Sorry for the detail. Core middleware has a rule to return named functions, so please fix that.

For example, this
https://github.com/honojs/hono/blob/7a366a518c2c92d31fcf26a846682ecb01ac9e97/src/middleware/pretty-json/index.ts#L42


```suggestion
  return async function contextStorage(c, next) {
```